### ISSUE TITLE
[generator] Performance regression?

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#6737]
My local map generation times increased significantly for some cases after #6537

E.g. Luxembourg generation time increased x2.7 (from 16min to 43min).

The biggest time consumer is in-between following log lines:
(note a 2243s difference)
```
LOG TID(1) INFO      142.471 generator/search_index_builder.cpp:694 BuildSearchIndex(): End building search index, elapsed seconds: 6.81536
LOG TID(1) INFO      142.496 generator/search_index_builder.cpp:617 BuildSearchIndexFromDataFile(): Search index size = 3475111
LOG TID(1) INFO      2385.41 generator/search_index_builder.cpp:582 BuildAddressTable(): Saved streets entries number: 108166
LOG TID(1) INFO      2385.41 generator/search_index_builder.cpp:583 BuildAddressTable(): Saved places entries number: 626
LOG TID(1) INFO      2385.41 generator/search_index_builder.cpp:588 BuildAddressTable(): Matched addresses percent: 99.7881 Total: 109022 Missing: 231
LOG TID(1) INFO      2385.44 generator/search_index_builder.cpp:626 BuildSearchIndexFromDataFile(): Streets table size: 239447 ; Places table size: 88932
```

Before the PR it was 591s (so this step consumes x3.8 more time)
```
LOG TID(1) INFO      138.762 generator/search_index_builder.cpp:615 BuildSearchIndex(): End building search index, elapsed seconds: 6.42105
LOG TID(1) INFO      138.787 generator/search_index_builder.cpp:539 BuildSearchIndexFromDataFile(): Search index size = 3219382
LOG TID(1) INFO      729.683 generator/search_index_builder.cpp:507 BuildAddressTable(): Address: BuildingToStreet entries count: 108130
LOG TID(1) INFO      729.683 generator/search_index_builder.cpp:513 BuildAddressTable(): Address: Matched percent 99.9519 Total: 108182 Missing: 52
LOG TID(1) INFO      729.719 generator/search_index_builder.cpp:546 BuildSearchIndexFromDataFile(): Search address table size = 239159
```

But e.g. for Montenegro the difference is smaller: the total generation time increased just 10%, while the above mentioned step has increased by x3.

Is this performance decrease justified or it could be optimized?

FYI @vng

**Comment by vng** [comment]
Yes, House->Place index generation added.
Make sure that for your local generation FLAGS_threads_count is not set or FLAGS_threads_count == 0.
By default, for one country generator_tool should use all CPU cores and should be much faster.

**Comment by pastk** [comment]
Yes my setup uses all 4 cores and I can see this particular new "House->Place" stage uses them also, but it looks like the most time-consuming stage anyway (for Luxembourg it takes 85% of total generation time, for Montenegro 20%).

**Comment by biodranik** [comment]
@pastk can you profile it?

**Comment by pastk** [comment]
> Yes, House->Place index generation added. 

I think it should be a skippable stage (the whole search index generation actually).
ATM I usually generate with `--skip="Coastline,Ugc,Popularity,Srtm,Descriptions,Routing,RoutingTransit,MwmDiffs,Statistics"`, but the search index is not skippable though its not needed for e.g. styles testing.

**Comment by vng** [comment]
You can introduce it, search for "generate_search_index" flag in steps.py
